### ISSUE TITLE
Trying to install the latest protobuf go library

### ISSUE THREAD

**Issue by PeterBocan** [thought-machine/please#1214]
I have been struggling the whole day installing the latest protobuf go library and resolving dependencies.
For unknown reason the following does not work:
```
go_get(
    name = "protobuf-fieldsort",
    get = "google.golang.org/protobuf/internal/fieldsort/...",
    revision = "v1.25.0",
)
```
getting the following is the error output:
```
 $ plz build //app/proto:proto
Build stopped after 3.66s. 1 target failed:
    //third_party/go:_protobuf-fieldsort#get
Error building target //third_party/go:_protobuf-fieldsort#get: exit status 1
go: warning: "google.golang.org/protobuf/internal/fieldsort/..." matched no packages
bash: line 0: cd: src/google.golang.org/protobuf/internal/fieldsort: No such file or directory
```

but the following code does work: 
```
go_get(
    name = "protobuf-fieldsort",
    get = "github.com/protocolbuffers/protobuf-go/internal/fieldsort/...",
    revision = "v1.25.0",
)
```

Thanks!
Peter.

**Comment by Tatskaari** [comment]
Hi Peter, 

I admit this is strange. It seems that `https://github.com/protocolbuffers/protobuf-go/tree/master/internal/fieldsort` doesn't exist on master. It does exist at [1.25.0](https://pkg.go.dev/mod/google.golang.org/protobuf@v1.25.0?tab=packages) though so in theory both of those should work. 

Perhaps we're not handling revisions correctly when going through a go module proxy. We don't use go modules for go_get yet. I will investigate. 

Thanks,

Jon

**Comment by Tatskaari** [comment]
Hi Peter,

TL;DR, vanity imports confuse our go_get rules a bit. You can do this instead:
```python
go_get(
    name = "protobuf-fieldsort",
    get = "google.golang.org/protobuf/...",
    install = ["internal/fieldsort"],
    revision = "v1.25.0",
)
```

The problem is we don't use go modules so to get a specific version, we cd into the directory and `git checkout -q $REVISION`. In this case the directory was `google.golang.org/protobuf/internal/fieldsort` because that's what your get was. `go get` downloads the latest version so that directory didn't exist.
